### ISSUE TITLE
[Windows|macOS|Linux|BSD] - Error when setting up a MIDI device

### ISSUE THREAD

**Issue by Drillimation** [rerrahkr/BambooTracker#293]
> #### Checklist
> 
> 
> 

---

## Bug Description

The error "Could not initialize MIDI input," followed by the text "MidiInWinMM::openPort: error creating Windows MM MIDI input port" appears when attempting to set up a MIDI device in BambooTracker.

## How to reproduce

1. Open up Configuration and go under the Sound tab
2. Under MIDI input, choose Windows MultiMedia and the device.
3. Click OK or Apply.

## System Information

- **Operating System**: Windows 10
- **BambooTracker Version**: v0.4.5

**Comment by OPNA2608** [comment]
I don't have an active Windows system to reproduce this so here's some purely theoretical help. The error is printed from the MIDI library we're using.

https://github.com/rerrahkr/BambooTracker/blob/af55ecb8951b63467f996085e0b7bc65c5994dd4/BambooTracker/midi/RtMidi/RtMidi.cpp#L2566-L2576

This sadly doesn't give us any indication of what the actual error code from the Windows API is, just that there was *some* error while requesting a connection through Windows's Winmm.lib library. [Here's a list](https://docs.microsoft.com/en-us/windows/win32/api/mmeapi/nf-mmeapi-midiinopen#return-value) of possible return codes from Microsoft's docs with fancy tech words to describe their situations, but I think `MMSYSERR_ALLOCATED` and `MMSYSERR_NOMEM` look promising in your case. Those might be returned when the port in question is already allocated/opened by another application.

---

tl;dr: Do you happen to have any other MIDI-capable application running that's connected to that device's MIDI-IN port? If so, could you try to close them and retry the Apply button? If not, does the MIDI-IN port work in another application in general?
